apps/omg/lib/omg/state/core.ex

```diff
@@ -16,9 +16,37 @@ defmodule OMG.State.Core do
   @moduledoc """
   The state meant here is the state of the ledger (UTXO set), that determines spendability of coins and forms blocks.
   All spend transactions, deposits and exits should sync on this for validity of moving funds.
+
+  We experienced long startup times on large UTXO set, which in some case caused timeouts and lethal `OMG.State`
+  restart loop. To mitigate this issue we introduced loading UTXO set on demand (see GH#1103) instead of full load
+  on process startup.
+
+  During OMG.State startup no UTXOs are fetched from DB, which is no longer blocking significantly.
+  Then during each of 6 utxo-related operations (see below) UTXO set is extended with UTXOs from DB to ensure operation
+  behavior hasn't been changed.
+
+  Transaction processing is populating the in-memory UTXO set and once block is formed newly created UTXO are inserted
```

Comment:
side question:
```
Transaction processing is populating the in-memory UTXO set and once block is formed newly created UTXO are inserted to DB, but are also kept in process State.
```
what happens if an outage happens in between? Between block is formed - newly created UTXOs are inserted to DB?
1. What if you're not able to submit a block, but were able to persist to DB.
2. What if you're able to submit a block, but not able to persist to DB.

Reply:
Case 2. is impossible. I'll describe case 1:

There are 2 processes engaged to block publishing
* OMG.State - check `form_block`
* OMG.ChildChain.BlockQueue - it check what gets mined on Ethereum and retries if needed

All _artifacts_ (mempool txs, new utxo, spends, block) from the OMG.State form_block is inserted to DB in single transaction (`multi_update`).
Then it publishes the event that new block is available - BQ is listening on these events.
If the outage blocks the event :point_up: propagation, BQ on init will check what gets mined on Ethereum and what block we have in DB and resend the missings.

Am I answered your concern?